### PR TITLE
Creating Satory Client for apollo + setup routes + cookie + Layout

### PR DESCRIPTION
**Pull request by Alexis-Chiche**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+NODE_PATH="src/"
+REACT_APP_HOST_GRAPHQL_development="http://localhost:3001/"
```
Comment (GreeFine) [inline .env]: You should not copy past things you not use

File: src/apollo/mutation/mutation.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  mutation login($password: String!, $username: String!) {
+    login(password: $password, username: $username) {
+      id
+      username
+      role
+    }
+  }
+`;
+
+export const CREATE_USER = gql`
```
Comment (GreeFine) [inline src/apollo/mutation/mutation.js]: You can leave `CREATE_USER` but you could change to REGISTER to match with the naming

File: src/apollo/query/query.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      published
+      author {
+        username
+        role
+      }
+    }
+  }
+`;
+
+export const GET_POST2 = gql`
```
Comment (GreeFine) [inline src/apollo/query/query.js]: Duplicate Code ?

File: src/components/inputs/InputUsername.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  function handleKeyPress(e) {
+    setErrForm(false);
+    setUsernameState({ value: e.target.value, error: false });
+  }
+
+  function validateUsername(e) {
+    const username = e.target.value;
+    const pattern = /[a-zA-Z][a-zA-Z0-9-_]{3,32}/g;
+
+    if (pattern.test(username)) setUsernameState({ value: username, error: false });
```
Comment (GreeFine) [inline src/components/inputs/InputUsername.jsx]:
```suggestion
    setUsernameState({ value: username, error: !pattern.test(username) });
```

File: src/components/inputs/InputUsername.jsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    else setUsernameState({ value: username, error: true });
```
Comment (GreeFine) [inline src/components/inputs/InputUsername.jsx]:
```suggestion
```

File: src/components/inputs/InputUsername.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+    console.log(usernameState);
```
Comment (GreeFine) [inline src/components/inputs/InputUsername.jsx]: No console.log in master

File: src/components/submitButton/LoginSubmit.jsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { objectOf, object, func, shape } from 'prop-types';
+import Cookies from 'universal-cookie';
+import { withRouter } from 'react-router-dom';
+import { Button } from 'reactstrap';
+import { useMutation } from '@apollo/react-hooks';
+import { LOGIN } from '../../apollo/mutation/mutation';
+
+const cookie = new Cookies();
+const FOURTEEN_DAYS = 1209600;
```
Comment (GreeFine) [inline src/components/submitButton/LoginSubmit.jsx]: #7

File: src/components/submitButton/RegisterSubmit.jsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      history.push('/login');
+    },
+    onError: () => {
+      setErrForm(true);
+    }
+  });
+
+  async function onSubmit() {
+    if (usernameState.value === '' || passwordState.value === '') setErrForm(true);
+    else if (usernameState.error === true || passwordState.error === true) setErrForm(true);
```
Comment (GreeFine) [inline src/components/submitButton/RegisterSubmit.jsx]: `usernameState.error === true` is redondant, usernameState.error is not a user input

File: src/components/inputs/InputUsername.jsx (hunk 6 above)
Comment (Alexis-Chiche) [reply src/components/inputs/InputUsername.jsx]: We're in dev.

File: src/components/submitButton/RegisterSubmit.jsx (hunk 8 above)
Comment (Alexis-Chiche) [reply src/components/submitButton/RegisterSubmit.jsx]: Must check value then error. At the rendering of the comp, value : '' and error: false
Thus, a void value should set the error on the form.

**Pull request merged**